### PR TITLE
API provides article creation functionality

### PR DESCRIPTION
**Pull request by ghost**
# [PT Story](https://www.pivotaltracker.com/story/show/171811237)
## Changes proposed in this pull request:
* Adds function in controller to create a new article and save it
* Adds happy and sad paths for tests
## What I have learned working on this feature:
* methods in binding.pry
* Divide tests with before blocks into separate describe blocks

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
-      resources :articles, only:[:index, :show]
+      resources :articles, only:[:index, :create]
+      post 'articles/create'
```
Comment (janko101) [inline config/routes.rb]: Try to remove this line and see if it works. I think we should have the post route open without it

File: spec/requests/api/v1/articles_can_be_created_spec.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        content: "content text"
+      }
+    end
+
+    it 'displays error of incomplete article' do
+      expect(response.status).to eq 422
+    end
+    
+    it 'displays error message on empty title' do
+      expect(JSON.parse(response.body)['message']).to eq "Content can't be blank"
```
Comment (janko101) [inline spec/requests/api/v1/articles_can_be_created_spec.rb]: Like we talked earlier, ping Thomas to see why we're not getting the correct error meassge

File: config/routes.rb (hunk 1 above)
Comment (hmvitous) [reply config/routes.rb]: We removed it and the tests still passed. 
So we are going to keep it out for the final pr.

**Review by Carltesio — COMMENTED** [review]
Your code looks ok for me

**Review by dernathul — COMMENTED** [review]
it seems ok to me.
good job guys

**Review by pgaunitz — APPROVED** [review]

**Review by dernathul — APPROVED** [review]

**Review by pgaunitz — APPROVED** [review]

**Review by emtalen — APPROVED** [review]

**Pull request merged**